px.py

```diff
@@ -832,7 +884,9 @@ def do_CONNECT(self):
                                 # No data means connection closed by remote host
                                 dprint("Connection closed by %s" % source)
                                 # Because tunnel is closed on one end there is no need to read from both ends
-                                rlist.clear()
+                                # rlist.clear() - stick with 2.7 compatible list operator.
+                                del rlist[:]
+                                
```

Comment:
Why is this better?

Reply:
.clear() is not available in python 2.7(?) -> error.
del rlist[:] does the same according to google? - but I am no python guru :).
(I happened to have 2.x installed for other reasons;  so saw the error.  But soon installed 3.x for it's advantages.  - but then had to reinstate 2.x for the 'other reasons' not working with 3.x -> now have both but need to swap between them.)